homeassistant/components/light/zha.py

```diff
@@ -32,9 +37,36 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     except (AttributeError, KeyError):
         pass
 
+    if discovery_info.get('color_capabilities') is None:
+        # ZCL Version 4 devices don't support the color_capabilities attribute.
+        # In this version XY support is mandatory, but we need to probe to
+        # determine if the device supports color temperature.
+        discovery_info['color_capabilities'] = CAPABILITIES_COLOR_XY
+        result = yield from safe_read(
+            endpoint.light_color, ['color_temperature'])
```

Comment:
This breaks any lights which do not have a light_color attribute. I'm putting a fix together now.

Reply:
Ah damn. My mistake.

Probably what we should do is move most of this logic out of the constructor and into setup. Then this could be properly nested under the color cluster logic:

https://github.com/home-assistant/home-assistant/blob/079d4039a10b819a0378580f7cb749959fe8ed43/homeassistant/components/light/zha.py#L83-L96